core/src/main/scala/org/apache/spark/deploy/RPackageUtils.scala

```diff
@@ -231,8 +236,10 @@ private[deploy] object RPackageUtils extends Logging {
     val zipOutputStream = new ZipOutputStream(new FileOutputStream(zipFile, false))
     try {
       filesToBundle.foreach { file =>
-        // get the relative paths for proper naming in the zip file
-        val relPath = file.getAbsolutePath.replaceFirst(dir.getAbsolutePath, "")
+        // Get the relative paths for proper naming in the zip file. Note that
+        // the separator should always be / for according to ZIP specification.
+        // `relPath` here should be, for example, "/packageTest/def.R" or "/test.R".
+        val relPath = file.toURI.toString.replaceFirst(dir.toURI.toString.stripSuffix("/"), "")
```

Comment:
Yeah I think using `/` always is good. Could you write a small comment on what the `toURI` is accomplishing here (as opposed to the the `getAbsolutePath` we were using before)